code-challenges.test.js

```diff
@@ -12,33 +12,174 @@
+// FAIL  ./code-challenges.test.js
+// words
+//   ✓ takes in a string and returns a string with a coded message, The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0 (1 ms)
+// findLetter
+//   ✕ returns an array of words containing the specified letter (case-insensitive)
+
+// ● findLetter › returns an array of words containing the specified letter (case-insensitive)
+
+//   ReferenceError: findLetter is not defined
+
+//     125 | describe('findLetter', () => {
+//     126 |   it('returns an array of words containing the specified letter (case-insensitive)', () => {
+//   > 127 |     expect(findLetter(fruitArray, "a")).toEqual(["Mango", "Apricot", "Peach"])
+//         |     ^
+//     128 |     expect(findLetter(fruitArray, "a").length).toEqual(3)
+//     129 |     expect(findLetter(fruitArray, "a")).toContain("Apricot")
+//     130 |   })
+
+//     at Object.expect (code-challenges.test.js:127:5)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 passed, 2 total
 
+//   ReferenceError: findLetter is not defined
+
+const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
 const filterLetterA = "a"
 // Expected output: ["Mango", "Apricot", "Peach"]
 const filterLetterE = "e"
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
+describe('findLetter', () => {
+  it('returns an array of words containing the specified letter (case-insensitive)', () => {
+    expect(findLetter(fruitArray, "a")).toEqual(["Mango", "Apricot", "Peach"])
+    expect(findLetter(fruitArray, "a").length).toEqual(3)
+    expect(findLetter(fruitArray, "a")).toContain("Apricot")
+  })
+})
+
+
+const findLetter = (array, letter) => {
+    return array.filter(word => word.toLowerCase().includes(letter.toLowerCase()))
+  }
+  
+  const findLetter2 = (array2, letter) => {
+    return array2.filter(word => word.toLowerCase().includes(letter.toLowerCase()))
+  }
 // b) Create the function that makes the test pass.
 
+// PASS  ./code-challenges.test.js
+// words
+//   ✓ takes in a string and returns a string with a coded message, The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0 (1 ms)
+// findLetter
+//   ✓ returns an array of words containing the specified letter (case-insensitive)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+
+
+
+
+
+
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
-// a) Create a test with expect statements using the variable provided.
+//Psuedocode:  I had to get help with this one, I can understand why I would use a for loop to iterste over the array of numbers but the rest of the code I am not sure how GPT came up with. I also had to get GPT to help with the test. My blocker is:  
+        // const num = hand[i];
+    //   counts[num] = (counts[num] || 0) + 1
```

Comment:
So you start by creating an empty object `counts`. Then you make a for loop to iterate through the input array. For each iteration, you are making a key value pair in that object. `counts[num]` is trying to access the key of `num` in the object `counts`. The logic here is saying create a key, which is `num` (which is `hand[i]`, and if it exists, the value for that key is going to be whatever it is + 1, OR, if it does not exist, then I want the value to be 0 + 1.